include/laser_filters/angular_interval_filter.h

```diff
@@ -0,0 +1,223 @@
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+#ifndef LASER_SCAN_ANGULAR_INTERVAL_FILTER_H
+#define LASER_SCAN_ANGULAR_INTERVAL_FILTER_H
+/**
+\author Ulas Sureyya Bingol
+@b ScanAngularIntervalFilter takes input scans and filters the readings within the ranges of interval sets.
+**/
+
+
+#include "filters/filter_base.h"
+#include "sensor_msgs/LaserScan.h"
+#include <laser_filters/AngularIntervalFilterConfig.h>
+#include <dynamic_reconfigure/server.h>
+
+namespace laser_filters
+{
+
+class LaserScanAngularIntervalFilter : public filters::FilterBase<sensor_msgs::LaserScan>
+{
+public:
+  std::string interval_sets_;
+  std::vector<std::vector<double>> intervals_;
+  std::shared_ptr<dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>> dyn_server_;
+  AngularIntervalFilterConfig param_config;
+  boost::recursive_mutex own_mutex_;
+
+  bool enable_ = true;
+  bool got_intervals_ = false;
+  bool configure()
+  {
+    ros::NodeHandle private_nh("~" + getName());
+    dyn_server_.reset(new dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>(own_mutex_, private_nh));
+    dynamic_reconfigure::Server<laser_filters::AngularIntervalFilterConfig>::CallbackType f;
+    f = boost::bind(&laser_filters::LaserScanAngularIntervalFilter::reconfigureCB, this, _1, _2);
+    dyn_server_->setCallback(f);
+
+
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("interval_sets"), interval_sets_))
+    {
+      ROS_WARN("AngularIntervalFilter was not given interval sets.\n");
+    }
+    else
+    {
+      param_config.interval_sets = interval_sets_;
+      dyn_server_->updateConfig(param_config);
+    }
+    return true;
+  }
+
+  virtual ~LaserScanAngularIntervalFilter()
+  {
+
+  }
+
+  bool update(const sensor_msgs::LaserScan& input_scan, sensor_msgs::LaserScan& filtered_scan)
+  {
+    filtered_scan = input_scan;
+    double replacement_value = std::numeric_limits<double>::quiet_NaN();
+    double current_angle = input_scan.angle_min;
+    if(got_intervals_ && enable_)
```

Comment:
Consider making the name of this variable a bit more detailed. Maybe `have_valid_intervals`?